config/tags.php

```diff
@@ -292,6 +293,7 @@
 				$attrs['controls'] = Str::toType($tag->controls ?? true, 'bool');
 				$attrs['loop']     = Str::toType($tag->loop, 'bool');
 				$attrs['muted']    = Str::toType($tag->muted ?? $autoplay, 'bool');
+				$attrs['playsinline'] = Str::toType($tag->playsinline ?? $autoplay, 'bool');
```

Comment:
> Breaking changes
> This changes how (video: my-video.mp4 autoplay: true) is rendered, so technically, this is a breaking change. That means that a video will be played inline (and not in full screen mode) on mobile user agents such as iOS and Android.

So when `autoplay` is enabled, should the `playsinline` be enabled too?

If we don't bind the `playsinline` attribute to `autoplay`, I guess there will be no breaking change if `playsinline` manual is enabled, right?

Reply:
> So when autoplay is enabled, should the playsinline be enabled too?

From the perspective of a rather non-technical user, the expectation is that adding `(video: my-video.mp4 autoplay: true)` will actually render a video that starts playback automatically. However, that is not the case (at least for iOS users), as in addition to the `autoplay` attribute  `playsinline` and `muted` are required in order to actually enable automatic playback on iOS.

> If we don't bind the playsinline attribute to autoplay, I guess there will be no breaking change if playsinline manual is enabled, right?

Yes, that’s true! I do think that defaulting to the value of the `autoplay` is sensible though for the UX reason outlined above. Also, the same logic is already implemented for the `muted` attribute, and I think it would be nice for both attributes to behave consistently.

Reply:
I'm not yet familiar with this new attribute. Does it still allow users to open full-screen mode when they want to really watch the video and listen to the audio? Or does it suppress that completely?

Reply:
Hi @distantnative, as long as the `controls` attribute is present (the video Kirbytag renders it by default), users have access to all the default playback controls, including volume and fullscreen toggles.

The video below shows the behavior on iOS for this HTML:

```html
<video autoplay muted playsinline controls width="100%">
  <source type=video/mp4 src=https://storage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4 />
</video>
```

https://user-images.githubusercontent.com/1512805/186371452-1d382560-1c1c-499d-92a4-9034f50ecf2d.mov

Reply:
Ok then I think it is a small breaking change but one that makes a lot of sense to me. I'd suggest to include it into 3.8

Reply:
Forgot to mention one more thing: On desktop browsers, `playsinline` doesn’t have any effect at all, i.e. it does not break existing behavior, as all major desktop browser play videos inline by default:

> The absence of the `playsinline` attribute does not imply that the video will display fullscreen by default. Indeed, most user agents have chosen to play all videos inline by default, and in such user agents the `playsinline` attribute has no effect.

https://html.spec.whatwg.org/multipage/media.html#attr-video-playsinline

To summarize:

* On desktop browsers, videos are dispalyed inline by default, there’s no breaking change.
* I’ve also tested this in Chrome and Firefox on Android 11. Even without the `playsinline` attribute, the video played inline, i.e. adding the `playsinline` attribute won’t break anything for these browsers.
* On iOS, `(video: my-video.mp4 autoplay: true)` previously had no effect because of the missing `playsinline` attribute and this change will enable automatic playback as expected.